src/api/creatures.py

```diff
@@ -103,182 +132,184 @@ def feed_creature(user_id: int, creature_id: int, treat_sku: str):
+                            COALESCE(quantity, 0) AS qty
+                        FROM treats
+                        JOIN users
+                            ON users.id = :id
+                            AND sku = :sku
+                        LEFT JOIN user_inventory_view AS inv
+                            ON inv.treat_sku = sku
+                            AND inv.user_id = users.id
+                        """
                     ),
                     {"sku": treat_sku, "id": user_id},
                 )
                 .mappings()
-                .fetchone()
+                .one_or_none()
             )
 
-            print(f"Inventory: {inventory}")
-
             if not inventory:
-                message = "This treat is not in your inventory"
+                raise HTTPException(
+                    status_code=404,
+                    detail=f"User {user_id} and/or treat {treat_sku} does not exist.",
+                )
+            if inventory["qty"] == 0:
+                raise HTTPException(
+                    status_code=403,
+                    detail=f"No '{treat_sku}'s found in inventory.",
+                )
+
+            print(f"[feed_creature] User {user_id}'s inventory:", inventory)
 
             stats = (
                 connection.execute(
                     sqlalchemy.text(
                         """
-                    SELECT 
-                        fav_treat, 
-                        hated_treat, 
-                        name,
-                        happiness,
-                        max_happiness, 
-                        max_hunger, 
-                        hunger,
-                        COALESCE(user_creature_connection.affinity, 0) AS affinity
-                    FROM creatures 
-                    JOIN creature_types 
-                        ON creatures.type = creature_types.type 
-                        AND creatures.id = :creature
-                    JOIN evolution_stages e
-                        ON creatures.stage = e.stage
-                    LEFT JOIN user_creature_connection
-                        ON creatures.id = user_creature_connection.creature_id 
-                    AND user_id = :user_id
-                """
+                        SELECT
+                            name,
+                            fav_treat,
+                            hated_treat,
+                            happiness,
+                            max_happiness,
+                            max_hunger,
+                            hunger,
+                            COALESCE(conn.affinity, 0) AS affinity
+                        FROM creatures
+                        JOIN creature_types
+                            ON creatures.type = creature_types.type
+                            AND creatures.id = :creature
+                        JOIN evolution_stages e
+                            ON creatures.stage = e.stage
+                        LEFT JOIN user_creature_connection AS conn
```

Comment:
This doesn't need to be changed, but fyi the "AS" is unnecessary. This line could be written as "LEFT JOIN user_creature_connection conn"

Reply:
yea, I'm aware. for consistency across all queries, I just chose to put AS for explicitness